stan/math/fwd/core/fvar.hpp

```diff
@@ -83,35 +85,14 @@ struct fvar {
    * Construct a forward variable with the specified value and
    * zero tangent.
    *
-   * @tparam V type of value (must be assignable to the value and
-   *   tangent type T)
+   * @tparam V type of value (must be assignable to T)
+   * @tparam Dummy dummy type inferred as part of enable-if
    * @param[in] v value
    */
-  fvar(const T& v) : val_(v), d_(0.0) {  // NOLINT(runtime/explicit)
-    if (unlikely(is_nan(v))) {
-      d_ = v;
-    }
-  }
-
-  /**
-   * Construct a forward variable with the specified value and
-   * zero tangent.
-   *
-   * @tparam V type of value (must be assignable to the value and
-   *   tangent type T)
-   * @param[in] v value
-   * @param[in] dummy value given by default with enable-if
-   *   metaprogramming
-   */
-  template <typename V>
-  fvar(const V& v,
-       typename std::enable_if<ad_promotable<V, T>::value>::type* dummy
-       = nullptr)
-      : val_(v), d_(0.0) {
-    if (unlikely(is_nan(v))) {
-      d_ = v;
-    }
-  }
+  template <typename V,
+            typename std::enable_if<ad_promotable<V, T>::value>::type* Dummy
+            = nullptr>
+  fvar(const V& v) : val_(v), d_(0) {}  // NOLINT(runtime/explicit)
```

Comment:
Is there a reason why this could not just be:

```cpp
  template <typename V,
            typename = std::enable_if_t<ad_promotable<V, T>::value>>
  fvar(const V& v) : val_(v), d_(0) {}  // NOLINT(runtime/explicit)
```